app/models/service_account.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: service_accounts
+#
+#  id          :bigint           not null, primary key
+#  uid         :string(255)      not null
+#  provider_id :bigint           unsigned, not null
+#  email       :string(255)      not null
+#  role        :string(255)      default("service_account"), not null
+#  level       :integer          default(0), not null
+#  state       :string(255)      default("pending"), not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+
+
+# ServiceAccount model
+class ServiceAccount < ApplicationRecord
+  belongs_to :user, foreign_key: "owner_id"
+  has_many :api_keys, as: :key_holder_account, dependent: :destroy, class_name: 'APIKey'
+
+  validate :role_exists
+  validate :email_n_uid_uniqueness
+  validates :email,       email: true, presence: true, uniqueness: true
+  validates :uid,         presence: true, uniqueness: true
+
+  scope :active, -> { where(state: 'active') }
+
+  after_update :disable_api_keys
+  before_create :assign_state
```

Comment:
Maybe we need to disable all service accounts if user account has been disabled?